gctree/branching_processes.py

```diff
@@ -987,3 +964,253 @@ def f(x):
     if not result.success:
         warnings.warn("optimization not sucessful, " + result.message, RuntimeWarning)
     return result.x[0], result.x[1]
+
+
+def _lltree(cm_counts, p: np.float64, q: np.float64) -> Tuple[np.float64, np.ndarray]:
```

Comment:
Do we need this function? Can't its contents define the existing `CollapsedTree.ll` method?

Reply:
We do need this, since computing likelihood on the hDAG bypasses CollapsedTree altogether, but does require the machinery in this function.